### DISCUSSION TITLE
Git Bash in Mingit.

### DISCUSSION THREAD

**Discussion by debajyoti1990** [git-for-windows/git#4577] [Ideas]
Git Bash is available in GUI version. But I think most users would really appreciate bash in mingit also.

**Comment by dscho** [comment]
The _whole point_ of MinGit is to reduce the size by leaving out components not needed by third-party applications, though. See https://github.com/git-for-windows/git/wiki/MinGit.

So you want MinGit to become... full Git for Windows?

**Reply by debajyoti1990** [reply]
No, just Git Bash. GUI is bloat in my opinion.

**Reply by dscho** [reply]
But you want MinGit to become interactive.

**Reply by dscho** [reply]
Just to clarify: `git-bash.exe` calls Bash _in a MinTTY window_, which is a GUI. `mintty.exe` is excluded from MinGit because 3rd-party applications wishing to add Git functionality via MinGit simply won't need MinTTY.

In short, granting the request would diminish the usefulness (and the clarity of purpose) of MinGit.

@debajyoti1990 if you want an extended version of MinGit, you can always build it yourself by forking https://github.com/git-for-windows/build-extra/, modifying [`mingit/release.sh`](https://github.com/git-for-windows/build-extra/blob/main/mingit/release.sh) accordingly and then running it in [a full Git for Windows SDK](https://gitforwindows.org/#download-sdk). Just please refrain from trying to put any of the maintenance burden of that extended version on anybody else.

**Reply by debajyoti1990** [reply]
Thanks for explaining. I will close the discussion.
